### PR TITLE
allow defining `SENTRY_SDK_NAME` externally

### PR DESCRIPTION
**Pull request by bruno-garcia**
Is this a feasible solution to this issue? https://github.com/getsentry/sentry-java/issues/1901

That assumes we can define `SENTRY_SDK_NAME` in the NDK [CMakeLists](https://github.com/getsentry/sentry-java/blob/main/sentry-android-ndk/CMakeLists.txt#L8) or [Gradle](https://github.com/getsentry/sentry-java/blob/33aeeb081e38aac8da02fd4101b6dfeb25d424ce/sentry-android-ndk/build.gradle.kts#L31-L32)

### CODE REVIEW

File: include/sentry.h [hunk 1]
```diff
@@ -23,7 +23,9 @@ extern "C" {
 #endif
 
 /* SDK Version */
+#ifndef SENTRY_SDK_NAME
 #define SENTRY_SDK_NAME "sentry.native"
```
Comment (bruno-garcia) [inline include/sentry.h:27]:
```suggestion
#    define SENTRY_SDK_NAME "sentry.native"
```

**Comment by bruno-garcia** [comment]
No change needed in the Android repo

**Comment by bruno-garcia** [comment]
I'll fix the tests during the week

**Comment by Mixaill** [comment]
Another approch would be something like

```cmake
#my_cool_project/Cmakelists.txt

...
set(SENTRY_SDK_NAME "another-sdk-name")
add_subdirectory(sentry-native)
...
```

```cmake
#my_cool_project/sentry-native/Cmakelists.txt

...
if(DEFINED SENTRY_SDK_NAME)
    target_compile_definitions(sentry PRIVATE SENTRY_SDK_NAME="${SENTRY_SDK_NAME}")
endif()
...
```

```c
//my_cool_project/sentry-native/include/sentry.h

...
#if !defined(SENTRY_SDK_NAME)
#    define SENTRY_SDK_NAME "sentry.native"
#endif
...
```

**Comment by bruno-garcia** [comment]
> Another approch would be something like. Minimal changes to the external projects. Also it does not require for external project to known anything about this project internals.
> 
> ```cmake
> #my_cool_project/Cmakelists.txt
> 
> ...
> set(SENTRY_SDK_NAME "another-sdk-name")
> add_subdirectory(sentry-native)
> ...
> ```
> 
> ```cmake
> #my_cool_project/sentry-native/Cmakelists.txt
> 
> ...
> if(DEFINED SENTRY_SDK_NAME)
>     target_compile_definitions(sentry PRIVATE SENTRY_SDK_NAME="${SENTRY_SDK_NAME}")
> endif()
> ...
> ```
> 
> ```c
> //my_cool_project/sentry-native/include/sentry.h
> 
> ...
> #if !defined(SENTRY_SDK_NAME)
> #    define SENTRY_SDK_NAME "sentry.native"
> #endif
> ...
> ```

On a second thought I believe it's best if it's not 'configurable' as we don't want custom values to be pass in here resulting in different variations. The Android case is a special one

**Review by marandaneto — APPROVED** [review]

**Review by Swatinem — APPROVED** [review]

**Comment by marandaneto** [comment]
@Swatinem can we cut a release to get this into the Android SDK? Thanks.

**Pull request merged**